src/main/groovy/com/bmuschko/gradle/docker/tasks/AbstractDockerRemoteApiTask.groovy

```diff
@@ -44,11 +43,23 @@ abstract class AbstractDockerRemoteApiTask extends DefaultTask {
 
     @TaskAction
     void start() {
-        threadContextClassLoader.withClasspath(getClasspath().files, createDockerClientConfig()) { dockerClient ->
+        runInDockerClassPath { dockerClient ->
             runRemoteCommand(dockerClient)
         }
     }
 
+    @InputFiles
+    public File[] getCertPathKeys() {
```

Comment:
This doesn't look related to the actual purpose of the change. I'd remove this for now and deal with it in a separate PR.